config/nginx.conf

```
@@ -13,21 +13,18 @@ server {
 
 server {
     listen 8063 default_server;
-    root /src;
 
-    location = /robots.txt {
-        alias /src/static/robots.txt;
+    root /src/frontends/mit-open/build;
+
+    location / {
+       try_files /static$uri $uri /index.html =404;
     }
 
     location = /.well-known/dnt-policy.txt {
         return 204;
     }
 
-    location = /favicon.ico {
-        try_files /static/images/favicon.ico /favicon.ico;
-    }
-
-    location / {
+    location ~ ^/(api|login|logout|admin|static/admin|static/hijack)/ {
```

Have I missed any routes for the web service here?

---

I'm not sure it's released yet, but there is an API http://localhost:8063/_features/. Possibly it should be api/v0. It's private insofar as it's a list of feature flags. On the other hand... the repository is public.

---

It's not ideal to have a fixed mapping, nginx provides the capability to avoid all of this anyway:
```suggestion
    location @proxy {
```

And the `try_files` directive above becomes:

```
try_files /static$uri $uri @proxy;
```

---

I was hoping for a better solution. The issue is that we need to serve the index.html at unknown routes for the frontend to load at all of its paths, so that needs to be the static file fallback. We could instead list all SPA handled first path parts, or list the backend's. This seemed the lesser of evils, though perhaps there's another option?

---

I've added `/_features` for the time being.

---

Ok, that's fair, let's leave this for now then. I guess at the very least the failure modes for the backend urls are more conspicuous.